gpytorch/lazy/cat_lazy_tensor.py

```diff
@@ -364,15 +364,24 @@ def devices(self):
     def device_count(self):
         return len(set(self.devices))
 
-    def to(self, device_id):
+    def to(self, device=None, dtype=None):
         """
-        returns a new CatLazyTensor with device_id as the output_device
+        returns a new CatLazyTensor with device as the output_device
         Warning: this does not move the LazyTensors in this CatLazyTensor to
-        device_id
+        device
         """
+        if type(device) is torch.dtype and dtype is None:
+            dtype = device
+            device = self.output_device
+
         new_kwargs = dict(self._kwargs)
-        new_kwargs["output_device"] = device_id
-        return self.__class__(*self._args, **new_kwargs)
+        new_kwargs["output_device"] = device
+        res = self.__class__(*self._args, **new_kwargs)
+
+        if dtype is not None:
+            res = res.type(dtype)
```

Comment:
This seems like it might be wasteful - We could instead add an `output_dtype` kwarg to the constructor and then handle things there?

Reply:
Wasteful as in defining too many new objects?

Reply:
Yeah. Probably not a big deal, but ideally we could construct `res` above directly with the right `dtype` rather than calling `.type` on it.